README.md
# PulpHUD

## What is PulpHUD ?
It's a simple modified HUD for Garry's Mod that I made on a friend's server for fun.
Lot of people on that server loved it so I put it on the workshop !

## Installation
The best and simple way is to open
http://steamcommunity.com/sharedfiles/filedetails/?id=148011719
and click on subscribe

## Configuration
Some cvars exists to config the hud at your will :
* **pulphud_enable 1/0** : Enable/disable the hud
* **pulphud_notification_sound_enable 1/0** : Enable/disable the sound of the notification when they pop in
* **pulphud_compass_enable 1/0** : Enable/disable the compass

## Changelog
Changelog is accessible [here](CHANGELOG.md "changelog")


CHANGELOG.md
# CHANGELOG

### Version 1.1.0rc
* (Fixed) Now tool gun will stop showing ammo
* (Added) New cvars & the possibility to change the HUD's color

### Version 1.0.0
* (Fixed) Now RPG and grenade will show ammo
* (Added) Blacklist to avoid AmmoHUD to show up
* (Fixed) Ammo showing when holding the gravity gun
* (Fixed) Compass not showing the correct north but it fucked up now
* (Improved) The HUD is now splitted into multiple files


lua/pulphud/sh_ammo.lua
-- Adding this file for the clientside
AddCSLuaFile()

if CLIENT then
	-- Settings
	PulpHUD_armorAnim = 0;
	PulpHUD_primaryAnim = 0;
	PulpHUD_secondaryAnim = 0;

	function PulpHUD_DrawAmmo()
		if PulpHUD_Enable:GetBool() == false then return; end

		-- Partie hud droite
		local offsetX = 10
		local offsetY = 10
		local w = 200
		local h = 80

		local ply = LocalPlayer()
		local wep = ply:GetActiveWeapon()

		local total_ammo = -1
		local primary_ammo = -1
		local secondary_ammo = -1
		if ply:Alive() and wep ~= nil and wep:IsWeapon() then
			if wep.GetPrimaryAmmoType ~= nil then
				total_ammo = ply:GetAmmoCount(wep:GetPrimaryAmmoType())
			end
			if wep.Clip1 ~= nil then
				primary_ammo = wep:Clip1()
			end
			if wep.GetSecondaryAmmoType ~= nil then
				secondary_ammo = ply:GetAmmoCount(wep:GetSecondaryAmmoType())
			end

			-- Blacklisting some weapon to avoid HUD from showingreload
			if wep:GetClass() == "weapon_physcannon" or wep:GetClass() == "weapon_physgun" or wep:GetClass() == "weapon_crowbar" or wep:GetClass() == "gmod_tool" then
				primary_ammo = -1
				total_ammo = -1
			end
		end

		-- This stuff fix the ammo for RPG and grenade (showing the HUD)
		local printTotal = true
		if primary_ammo < 0 and total_ammo >= 0 then
			primary_ammo = total_ammo
			printTotal = false
		end

		if primary_ammo >= 0 and PulpHUD_primaryAnim < w then
			PulpHUD_primaryAnim=PulpHUD_primaryAnim+5
		elseif primary_ammo == -1 and PulpHUD_primaryAnim > 0 and PulpHUD_secondaryAnim==0 then
			PulpHUD_primaryAnim=PulpHUD_primaryAnim-5
		end
		draw.RoundedBox( 8, ScrW()-PulpHUD_primaryAnim-offsetX, ScrH()-h-offsetY, w, h, PulpHUD_Color((PulpHUD_primaryAnim/w)*180) );
		draw.DrawText("Primary", "HealthFont", ScrW()-PulpHUD_primaryAnim-offsetX+10, ScrH()-h-offsetY+10, Color(255, 255, 255, (PulpHUD_primaryAnim/w)*255), TEXT_ALIGN_LEFT);
		draw.DrawText(primary_ammo, "HealthFont2", ScrW()-PulpHUD_primaryAnim-offsetX+80, ScrH()-h-offsetY+15, Color(255, 255, 255, (PulpHUD_primaryAnim/w)*255), TEXT_ALIGN_LEFT);
		if printTotal then
			draw.DrawText(total_ammo, "HealthFont", ScrW()-PulpHUD_primaryAnim-offsetX+w-10, ScrH()-offsetY-30, Color(255, 255, 255, (PulpHUD_primaryAnim/w)*255), TEXT_ALIGN_RIGHT);
		end

		if secondary_ammo > 0 and PulpHUD_secondaryAnim < 50 and PulpHUD_primaryAnim == w then
			PulpHUD_secondaryAnim=PulpHUD_secondaryAnim+1
		elseif secondary_ammo <= 0 and PulpHUD_secondaryAnim > 0 then
			PulpHUD_secondaryAnim=PulpHUD_secondaryAnim-1
		end
		draw.RoundedBox( 8, ScrW()-w-offsetX, ScrH()-h-offsetY-PulpHUD_secondaryAnim, w/1.5, 40, PulpHUD_Color( (PulpHUD_secondaryAnim/50)*180) );
		draw.DrawText("Secondary", "BudgetLabel", ScrW()-w-offsetX+10, ScrH()-h-offsetY+10-PulpHUD_secondaryAnim, Color(255, 255, 255, (PulpHUD_secondaryAnim/50)*255), TEXT_ALIGN_LEFT);
		draw.DrawText(secondary_ammo, "HealthFont", ScrW()-w-offsetX+80, ScrH()-h-offsetY+10-PulpHUD_secondaryAnim, Color(255, 255, 255, (PulpHUD_secondaryAnim/50)*255), TEXT_ALIGN_LEFT);
	end
	hook.Add("HUDPaint", "PulpHUD_DrawAmmo", PulpHUD_DrawAmmo)

	function PulpHUD_HideAmmo( name )
		if PulpHUD_Enable:GetBool() == false then return; end
		if (name == "CHudAmmo") or (name == "CHudSecondaryAmmo") then
	        return false
	    end
	end
	hook.Add( "HUDShouldDraw", "PulpHUD_HideAmmo", PulpHUD_HideAmmo )
end


lua/pulphud/sh_health.lua
-- Adding this file for the clientside
AddCSLuaFile()

if CLIENT then
	surface.CreateFont("HealthFont", {
	    font    = "BudgetLabel",
	    size    = 20,
	    weight  = 1000,
	    antialias = true,
	    shadow = false,
		outline = true
	})

	surface.CreateFont("HealthFont2", {
	    font    = "BudgetLabel",
	    size    = 50,
	    weight  = 1000,
	    antialias = true,
	    shadow = false,
		outline = true
	})

	local angleY = 0

	function PulpHUD_DrawHealth()
		if PulpHUD_Enable:GetBool() == false then return; end

		-- Partie HEALTH bar
		local offsetX = 10
		local offsetY = 10
		local w = 200
		local h = 80

		local pHealth = LocalPlayer():Health()
		local healthOffset = 80
		if pHealth <= 0 then
			pHealth = "DEAD"
			healthOffset = 70
		end
		draw.RoundedBox( 8, offsetX, ScrH()-h-offsetY, w, h, PulpHUD_Color(180) );
		draw.DrawText("Health", "HealthFont", offsetX+10, ScrH()-h-offsetY+10, Color(255, 255, 255, 255), TEXT_ALIGN_LEFT);
		draw.DrawText(pHealth, "HealthFont2", offsetX+healthOffset, ScrH()-h-offsetY+15, Color(255, 255, 255, 255), TEXT_ALIGN_LEFT);

		if LocalPlayer():Armor() > 0 and PulpHUD_armorAnim < 50 then
			PulpHUD_armorAnim=PulpHUD_armorAnim+1
		elseif LocalPlayer():Armor() == 0 and PulpHUD_armorAnim > 0 then
			PulpHUD_armorAnim=PulpHUD_armorAnim-1
		end
		draw.RoundedBox( 8, offsetX+w/2, ScrH()-h-offsetY-PulpHUD_armorAnim, w/2, 40, PulpHUD_Color( (PulpHUD_armorAnim/50)*180 ) );
		draw.DrawText("Armor", "BudgetLabel", offsetX+w/2+10, ScrH()-h-offsetY-PulpHUD_armorAnim+10, Color(255, 255, 255, (PulpHUD_armorAnim/50)*255), TEXT_ALIGN_LEFT);
		draw.DrawText(LocalPlayer():Armor(), "HealthFont", offsetX+w/2+60, ScrH()-h-offsetY-PulpHUD_armorAnim+10, Color(255, 255, 255, (PulpHUD_armorAnim/50)*255), TEXT_ALIGN_LEFT);
	end
	hook.Add("HUDPaint", "PulpHUD_DrawHealth", PulpHUD_DrawHealth)

	function PulpHUD_HideHealth( name )
		if PulpHUD_Enable:GetBool() == false then return; end
		if (name == "CHudHealth") or (name == "CHudBattery") then
	        return false
	    end
	end
	hook.Add( "HUDShouldDraw", "PulpHUD_HideHealth", PulpHUD_HideHealth )
end


lua/pulphud/sh_compass.lua
-- Adding this file for the clientside
AddCSLuaFile()

if CLIENT then
	-- Variables pour la boussole
	local PulpHUD_Var_CompassDraw = CreateClientConVar( "pulphud_compass_enable", "1" )

	surface.CreateFont("CompassFont", {
	    font    = "BudgetLabel",
	    size    = 25,
	    weight  = 1000,
	    antialias = true,
	    shadow = false,
		outline = true
	})

	PulpHUD_Compass = {}
	PulpHUD_Compass[90] = "N"
	PulpHUD_Compass[135] = "NW"
	PulpHUD_Compass[180] = "W"
	PulpHUD_Compass[-180] = "W"
	PulpHUD_Compass[-135] = "SW"
	PulpHUD_Compass[-90] = "S"
	PulpHUD_Compass[-45] = "SE"
	PulpHUD_Compass[0] = "E"
	PulpHUD_Compass[45] = "NE"
	for i=-180,180 do
		if not PulpHUD_Compass[i] then
			PulpHUD_Compass[i] = " "
		end
	end

	function PulpHUD_DrawCompass()
		if PulpHUD_Enable:GetBool() == false then return; end

		-- Compass
		if PulpHUD_Var_CompassDraw:GetBool() == false then return; end
		local w = ScrW()/3
		local h = 30
		draw.RoundedBox( 8, ScrW()/2-w/2, 100, w, h, PulpHUD_Color(180) );

		local finalText = ""
		local yaw = math.floor(LocalPlayer():GetAngles().y)
		for i=yaw,yaw+28,1 do
			local y=i
			if i>180 then
				y=i-180
			end
			finalText = PulpHUD_Compass[y]..finalText
		end
		for i=yaw-1,yaw-28,-1 do
			local y=i
			if i<-180 then
				y=180-math.abs(i)
			end
			finalText = finalText..PulpHUD_Compass[y]
		end
		draw.DrawText(finalText, "CompassFont", ScrW()/2, 100+2, Color(255, 255, 255, 255), TEXT_ALIGN_CENTER);
	end
	hook.Add("HUDPaint", "PulpHUD_DrawCompass", PulpHUD_DrawCompass)
end


lua/autorun/pulphud_loader.lua
-- To know which side, we are in the log :)
local SIDE = "UNKNOWN"
if SERVER then SIDE = "SERVERSIDE" end
if CLIENT then SIDE = "CLIENTSIDE" end
print("[PULPHUD] Loading " .. SIDE .. " ...")

-- Adding this file for the clientside
AddCSLuaFile()

-- Variables globales pour l'HUD
PulpHUD_Enable = CreateClientConVar( "pulphud_enable", "1" )

PulpHUD_Color_Red = CreateClientConVar( "pulphud_color_red", "0" )
PulpHUD_Color_Green = CreateClientConVar( "pulphud_color_green", "0" )
PulpHUD_Color_Blue = CreateClientConVar( "pulphud_color_blue", "0" )

function PulpHUD_Color(alpha)
    return Color(PulpHUD_Color_Red:GetInt(), PulpHUD_Color_Green:GetInt(), PulpHUD_Color_Blue:GetInt(), alpha)
end

-- Loading shared files
local sharedFilesToLoad = {"sh_health.lua", "sh_ammo.lua", "sh_notification.lua", "sh_compass.lua"}
for k,file in pairs(sharedFilesToLoad) do
	include("../pulphud/"..file)
end

print("[PULPHUD] Loaded successfully !")


lua/pulphud/sh_notification.lua
-- Adding this file for the clientside
AddCSLuaFile()

if CLIENT then
	-- Variables pour les notifications
	local PulpHUD_Notification_Enable = CreateClientConVar( "pulphud_notification_enable", "1" ) -- Pas encore implantée :(
	local PulpHUD_Notification_Sound_Enable = CreateClientConVar( "pulphud_notification_sound_enable", "1" )

	PulpHUD_notifs = {}
	function PulpHUD_addNotification(text)
		local data = {}
		data[1] = text -- texte de la notification
		data[2] = ScrW() -- position X
		data[3] = ScrH()/2 -- position Y
		data[4] = 0 -- vitesse X
		data[5] = 0 -- vitesse Y
		data[6] = 255 -- alpha
		data[7] = 0 -- vitesse alpha
		data[8] = 0 -- Timer pour eviter la disparition rapide
		data[9] = false -- Est apparue ?

		table.insert( PulpHUD_notifs , data );
	end

	timer.Simple(1.5, function()
		function GAMEMODE:AddNotify(Text, Icon, Timeout)
			PulpHUD_addNotification(Text)
		end
	end)

	function PulpHUD_DrawNotification()
		if PulpHUD_Enable:GetBool() == false then return; end

		-- Partie notification
		for i, data in pairs(PulpHUD_notifs) do
			if data[6] > 0 then
				surface.SetFont("BudgetLabel");
				local w,h = surface.GetTextSize(data[1]);
				w = w+32;
				if data[2] < ScrW() then
					draw.RoundedBox( 8, data[2], data[3], w, 20, PulpHUD_Color(data[6]-75) ); -- On dessine la boite
					draw.DrawText(data[1], "BudgetLabel", data[2]+24, data[3]+4, Color(255, 255, 255, data[6]), TEXT_ALIGN_LEFT); -- On dessine le texte
					draw.TexturedQuad
					{
						texture = surface.GetTextureID("gui/gmod_logo"),
						color = Color(255, 255, 255, data[6]),
						x = data[2],
						y = data[3],
						w = 24,
						h = 24
					}; -- On dessine l'icone

					if data[9] == false then
						if PulpHUD_Notification_Sound_Enable:GetBool() then
							surface.PlaySound( "garrysmod/content_downloaded.wav" );
						end
						data[9] = true;
					end
				end

				local canMoveX = true;
				local canMoveY = true;
				local last_notif = PulpHUD_notifs[i-1];
				if last_notif != nil then
					if data[3]+data[5] <= last_notif[3]+21 then
						canMoveX = false;
						data[4] = 0;
					end
					if data[3]+data[5] <= last_notif[3]+22 and last_notif[6] > 0 then
						canMoveY = false;
						data[5] = 0;
					end
				end

				-- Déplacement
				if canMoveX and data[2] > ScrW()-w then
					data[4] = data[4]-0.4;
					data[2] = data[2]+data[4];
				elseif canMoveY and data[3] > 200 then
					data[5] = data[5]-0.4;
					data[3] = data[3]+data[5];
				end

				-- Disparition en alpha
				if data[2] <= ScrW()-w and data[3] <= 200 then
					if data[8] >= 400 then
						data[7] = data[7]-1;
						data[6] = data[6]+data[7];
						if data[6] < 0 then
							data[6] = 0;
							data[7] = 0;
						end
					else
						data[8] = data[8]+1;
					end
				end

			end

		end -- FIN DE BOUCLE
	end
	hook.Add("HUDPaint", "PulpHUD_DrawNotification", PulpHUD_DrawNotification);
end


lua/autorun/client/hud.lua
print("Loading PulpHUD ...")

-- Variables globales pour l'HUD
local PulpHUD_Enable = CreateClientConVar( "pulphud_enable", "1" )

-- Variables pour les notifications
local PulpHUD_Notification_Enable = CreateClientConVar( "pulphud_notification_enable", "1" ) -- Pas encore implantée :(
local PulpHUD_Notification_Sound_Enable = CreateClientConVar( "pulphud_notification_sound_enable", "1" )

-- Variables pour la boussole
local PulpHUD_Var_CompassDraw = CreateClientConVar( "pulphud_compass_enable", "1" )

surface.CreateFont("HealthFont", {
    font    = "BudgetLabel",
    size    = 20,
    weight  = 1000,
    antialias = true,
    shadow = false,
	outline = true
})

surface.CreateFont("HealthFont2", {
    font    = "BudgetLabel",
    size    = 50,
    weight  = 1000,
    antialias = true,
    shadow = false,
	outline = true
})

surface.CreateFont("CompassFont", {
    font    = "BudgetLabel",
    size    = 25,
    weight  = 1000,
    antialias = true,
    shadow = false,
	outline = true
})

function PulpHUD_Drawer()
	if PulpHUD_Enable:GetBool() == false then return; end

	-- Partie notification
	for i, data in pairs(PulpHUD_notifs) do
		if data[6] > 0 then
			surface.SetFont("BudgetLabel");
			local w,h = surface.GetTextSize(data[1]);
			w = w+32;
			if data[2] < ScrW() then
				draw.RoundedBox( 8, data[2], data[3], w, 20, Color(0, 0, 0, data[6]-75) ); -- On dessine la boite
				draw.DrawText(data[1], "BudgetLabel", data[2]+24, data[3]+4, Color(255, 255, 255, data[6]), TEXT_ALIGN_LEFT); -- On dessine le texte
				draw.TexturedQuad
				{
					texture = surface.GetTextureID("gui/gmod_logo"),
					color = Color(255, 255, 255, data[6]),
					x = data[2],
					y = data[3],
					w = 24,
					h = 24
				}; -- On dessine l'icone

				if data[9] == false then
					if PulpHUD_Notification_Sound_Enable:GetBool() then
						surface.PlaySound( "garrysmod/content_downloaded.wav" );
					end
					data[9] = true;
				end
			end

			local canMoveX = true;
			local canMoveY = true;
			local last_notif = PulpHUD_notifs[i-1];
			if last_notif != nil then
				if data[3]+data[5] <= last_notif[3]+21 then
					canMoveX = false;
					data[4] = 0;
				end
				if data[3]+data[5] <= last_notif[3]+22 and last_notif[6] > 0 then
					canMoveY = false;
					data[5] = 0;
				end
			end

			-- Déplacement
			if canMoveX and data[2] > ScrW()-w then
				data[4] = data[4]-0.4;
				data[2] = data[2]+data[4];
			elseif canMoveY and data[3] > 200 then
				data[5] = data[5]-0.4;
				data[3] = data[3]+data[5];
			end

			-- Disparition en alpha
			if data[2] <= ScrW()-w and data[3] <= 200 then
				if data[8] >= 400 then
					data[7] = data[7]-1;
					data[6] = data[6]+data[7];
					if data[6] < 0 then
						data[6] = 0;
						data[7] = 0;
					end
				else
					data[8] = data[8]+1;
				end
			end

		end

	end -- FIN DE BOUCLE

	-- Partie hud gauche
	local offsetX = 10
	local offsetY = 10
	local w = 200
	local h = 80

	local pHealth = LocalPlayer():Health()
	local healthOffset = 80
	if pHealth <= 0 then
		pHealth = "DEAD"
		healthOffset = 70
	end
	draw.RoundedBox( 8, offsetX, ScrH()-h-offsetY, w, h, Color(0, 0, 0, 180) );
	draw.DrawText("Health", "HealthFont", offsetX+10, ScrH()-h-offsetY+10, Color(255, 255, 255, 255), TEXT_ALIGN_LEFT);
	draw.DrawText(pHealth, "HealthFont2", offsetX+healthOffset, ScrH()-h-offsetY+15, Color(255, 255, 255, 255), TEXT_ALIGN_LEFT);

	if LocalPlayer():Armor() > 0 and PulpHUD_armorAnim < 50 then
		PulpHUD_armorAnim=PulpHUD_armorAnim+1
	elseif LocalPlayer():Armor() == 0 and PulpHUD_armorAnim > 0 then
		PulpHUD_armorAnim=PulpHUD_armorAnim-1
	end
	draw.RoundedBox( 8, offsetX+w/2, ScrH()-h-offsetY-PulpHUD_armorAnim, w/2, 40, Color(0, 0, 0, (PulpHUD_armorAnim/50)*180) );
	draw.DrawText("Armor", "BudgetLabel", offsetX+w/2+10, ScrH()-h-offsetY-PulpHUD_armorAnim+10, Color(255, 255, 255, (PulpHUD_armorAnim/50)*255), TEXT_ALIGN_LEFT);
	draw.DrawText(LocalPlayer():Armor(), "HealthFont", offsetX+w/2+60, ScrH()-h-offsetY-PulpHUD_armorAnim+10, Color(255, 255, 255, (PulpHUD_armorAnim/50)*255), TEXT_ALIGN_LEFT);

	-- Partie hud droite
	local offsetX = 10
	local offsetY = 10
	local w = 200
	local h = 80

	local ply = LocalPlayer()
	local wep = ply:GetActiveWeapon()

	local total_ammo = -1
	local primary_ammo = -1
	local secondary_ammo = -1
	if ply:Alive() and wep ~= nil then
		if wep.GetPrimaryAmmoType ~= nil then
			total_ammo = ply:GetAmmoCount(wep:GetPrimaryAmmoType())
		end
		if wep.Clip1 ~= nil then
			primary_ammo = wep:Clip1()
		end
		if wep.GetSecondaryAmmoType ~= nil then
			secondary_ammo = ply:GetAmmoCount(wep:GetSecondaryAmmoType())
		end
	end

	if primary_ammo >= 0 and PulpHUD_primaryAnim < w then
		PulpHUD_primaryAnim=PulpHUD_primaryAnim+5
	elseif primary_ammo == -1 and PulpHUD_primaryAnim > 0 and PulpHUD_secondaryAnim==0 then
		PulpHUD_primaryAnim=PulpHUD_primaryAnim-5
	end
	draw.RoundedBox( 8, ScrW()-PulpHUD_primaryAnim-offsetX, ScrH()-h-offsetY, w, h, Color(0, 0, 0, (PulpHUD_primaryAnim/w)*180) );
	draw.DrawText("Primary", "HealthFont", ScrW()-PulpHUD_primaryAnim-offsetX+10, ScrH()-h-offsetY+10, Color(255, 255, 255, (PulpHUD_primaryAnim/w)*255), TEXT_ALIGN_LEFT);
	draw.DrawText(primary_ammo, "HealthFont2", ScrW()-PulpHUD_primaryAnim-offsetX+80, ScrH()-h-offsetY+15, Color(255, 255, 255, (PulpHUD_primaryAnim/w)*255), TEXT_ALIGN_LEFT);
	draw.DrawText(total_ammo, "HealthFont", ScrW()-PulpHUD_primaryAnim-offsetX+w-10, ScrH()-offsetY-30, Color(255, 255, 255, (PulpHUD_primaryAnim/w)*255), TEXT_ALIGN_RIGHT);

	if secondary_ammo > 0 and PulpHUD_secondaryAnim < 50 and PulpHUD_primaryAnim == w then
		PulpHUD_secondaryAnim=PulpHUD_secondaryAnim+1
	elseif secondary_ammo <= 0 and PulpHUD_secondaryAnim > 0 then
		PulpHUD_secondaryAnim=PulpHUD_secondaryAnim-1
	end
	draw.RoundedBox( 8, ScrW()-w-offsetX, ScrH()-h-offsetY-PulpHUD_secondaryAnim, w/1.5, 40, Color(0, 0, 0, (PulpHUD_secondaryAnim/50)*180) );
	draw.DrawText("Secondary", "BudgetLabel", ScrW()-w-offsetX+10, ScrH()-h-offsetY+10-PulpHUD_secondaryAnim, Color(255, 255, 255, (PulpHUD_secondaryAnim/50)*255), TEXT_ALIGN_LEFT);
	draw.DrawText(secondary_ammo, "HealthFont", ScrW()-w-offsetX+80, ScrH()-h-offsetY+10-PulpHUD_secondaryAnim, Color(255, 255, 255, (PulpHUD_secondaryAnim/50)*255), TEXT_ALIGN_LEFT);

	-- Compass
	if PulpHUD_Var_CompassDraw:GetBool() == false then return; end
	local w = ScrW()/3
	local h = 30
	draw.RoundedBox( 8, ScrW()/2-w/2, 100, w, h, Color(0, 0, 0, 180) );

	local finalText = ""
	local yaw = math.floor(LocalPlayer():GetAngles().y)
	for i=yaw,yaw+28,1 do
		local y=i
		if i>180 then
			y=i-180
		end
		finalText = PulpHUD_Compass[y]..finalText
	end
	for i=yaw-1,yaw-28,-1 do
		local y=i
		if i<-180 then
			y=180-math.abs(i)
		end
		finalText = finalText..PulpHUD_Compass[y]
	end
	draw.DrawText(finalText, "CompassFont", ScrW()/2, 100+2, Color(255, 255, 255, 255), TEXT_ALIGN_CENTER);
end
hook.Add("HUDPaint", "PulpHUD_Drawer", PulpHUD_Drawer);

-- Settings
PulpHUD_armorAnim = 0;
PulpHUD_primaryAnim = 0;
PulpHUD_secondaryAnim = 0;

PulpHUD_Compass = {}
PulpHUD_Compass[0] = "N"
PulpHUD_Compass[45] = "NW"
PulpHUD_Compass[90] = "W"
PulpHUD_Compass[135] = "SW"
PulpHUD_Compass[180] = "S"
PulpHUD_Compass[-180] = "S"
PulpHUD_Compass[-135] = "SE"
PulpHUD_Compass[-90] = "E"
PulpHUD_Compass[-45] = "NE"
for i=-180,180 do
	if not PulpHUD_Compass[i] then
		PulpHUD_Compass[i] = " "
	end
end


PulpHUD_notifs = {}
function PulpHUD_addNotification(text)
	local data = {}
	data[1] = text -- texte de la notification
	data[2] = ScrW() -- position X
	data[3] = ScrH()/2 -- position Y
	data[4] = 0 -- vitesse X
	data[5] = 0 -- vitesse Y
	data[6] = 255 -- alpha
	data[7] = 0 -- vitesse alpha
	data[8] = 0 -- Timer pour eviter la disparition rapide
	data[9] = false -- Est apparue ?

	table.insert( PulpHUD_notifs , data );
end

function PulpHUD_HideThings( name )
	if PulpHUD_Enable:GetBool() == false then return; end
	if(name == "CHudHealth") or (name == "CHudBattery") or (name == "CHudAmmo") or (name == "CHudSecondaryAmmo") then
        return false
    end
end
hook.Add( "HUDShouldDraw", "PulpHUD_HideThings", PulpHUD_HideThings )

timer.Simple(1.5, function()
	function GAMEMODE:AddNotify(Text, Icon, Timeout)
		PulpHUD_addNotification(Text)
	end
end)

print("PulpHUD loaded !")
